packages/dds/merge-tree/src/test/mergeTreeOperationRunner.ts

```diff
@@ -161,22 +161,22 @@ export function generateOperationMessagesForClients(
     for (let i = 0; i < opsPerRound; i++) {
         // pick a client greater than 0, client 0 only applies remote ops
         // and is our baseline
-        const client = clients[random.integer(1, clients.length - 1)(mt)];
+        const client = clients[random.integer(1, clients.length - 1)];
         const len = client.getLength();
         const sg = client.peekPendingSegmentGroups();
         let op: IMergeTreeOp | undefined;
-        if (len === 0 || len < minLength) {
-            const text = client.longClientId!.repeat(random.integer(1, 3)(mt));
+        if (len === 0 || len < minLength || (len < minLength * 3 && random.bool(1 / 3))) {
```

Comment:
why add the insert condition? can't the test just add an insert operation if it wants it?

Reply:
this is just meant to be a fallback case, that handles empty/min len, as we usually want data in the tree.

Reply:
yeah, that seems better. I'll convert tests that seem like they should have an insert op to inject one in config. The lack of inserts was one reason the reconnect farm didn't repro this issue (text was only inserted when the string dipped below min length, so it was pretty unlikely to have a lot of adjacent surviving segments that would end up in a different order on rebase)